### ISSUE TITLE
Left Nav links are not clickable on Windows

### ISSUE THREAD

**Issue by VolodymyrPliuta** [carbon-design-system/carbon#8917]
## Environment
Browser
> Operating system
Windows OS
> Browser
Chrome and FireFox v88.0.1 (64-bit)

## Detailed description
Bug is just on Windows OS, Mac OS with Voice Over handles it fine. We have a user with diversability, that uses screen reader.  when user tries to navigate to left nav and click links they are not clickable. also Screen reader don't see(can't read text values of <SideNavMenuItem/>) on the left nav. if I downgrade two package: to older versions: 
"carbon-components": "10.28.0",
"carbon-components-react": "7.28.0"
screen reader on Windows OS starts working fine.
> What version of the Carbon Design System are you using?
  "carbon-components": "^10.34.0",
  "carbon-components-react": "^7.34.0"
  
> What did you expect to happen?
Screen reader should work on Windows OS
> What happened instead?
Screen reader doesn't read left nav and links are not clickable when user press Enter

## Steps to reproduce the issue

1. Get on Windows OS
2. Turn on Screen reader
3. Try to navigate to left nav and click the link

> Please create a reduced test case in CodeSandbox
> - React:
>   https://codesandbox.io/s/header-with-sidenav-forked-7glvb?file=/src/index.js

**Comment by VolodymyrPliuta** [comment]
<!-- Feel free to remove sections that aren't relevant.

## Title line template: [Title]: Links are not clickable on Windows OS

## Environment
Browser
> Operating system
Windows OS
> Browser
Chrome and FireFox v88.0.1 (64-bit)

## Detailed description
Bug is just on Windows OS, Mac OS with Voice Over handles it fine. We have a user with diversability, that uses screen reader.  when user tries to navigate to left nav and click links they are not clickable. also Screen reader don't see(can't read text values of <SideNavMenuItem/>) on the left nav. if I downgrade two package: to older versions: 
"carbon-components": "10.28.0",
"carbon-components-react": "7.28.0"
screen reader on Windows OS starts working fine.
> What version of the Carbon Design System are you using?
  "carbon-components": "^10.34.0",
  "carbon-components-react": "^7.34.0"
  
> What did you expect to happen?
Screen reader should work on Windows OS
> What happened instead?
Screen reader doesn't read left nav and links are not clickable when user press Enter

## Steps to reproduce the issue

1. Get on Windows OS
2. Turn on Screen reader
3. Try to navigate to left nav and click the link

> Please create a reduced test case in CodeSandbox
> - React:
>   https://codesandbox.io/s/header-with-sidenav-forked-7glvb?file=/src/index.js

**Comment by VolodymyrPliuta** [comment]
with some reason, I don't see my issue description, but when I click edit it's there, can be a GitHub bug

**Comment by joshblack** [comment]
Hi there @VolodymyrPliuta! 👋  Could you share more about the screen reader being used? Is it maybe NVDA or JAWS that is being used?

**Comment by VolodymyrPliuta** [comment]
Hi @joshblack 👋 , screen reader NVDA(latest), OS - Windows 10(Latest)

**Comment by joshblack** [comment]
Sounds good! @VolodymyrPliuta Thanks so much. cc @dakahn to help check out what's going on here 👀

**Comment by dakahn** [comment]
Unable to recreate using latest NVDA on latest Firefox in Windows 10. I'm able to access the side nav, focus and operate various controls. Controls are read and identified properly. I'm also able to use the <kbd>d</kbd> key to navigate to the navigation as a page landmark. 

![jawspagination](https://user-images.githubusercontent.com/40970507/122947563-de3ef380-d33f-11eb-8ff3-8e7984b5a655.gif)

**Comment by joshblack** [comment]
Going to close this out since there seemed to not be a way to reproduce. If there are steps to reproduce the problem, please comment and we can re-open!
